modules/integration/src/test/scala/scala/cli/integration/PackageTestDefinitions.scala

```diff
@@ -72,6 +72,96 @@ abstract class PackageTestDefinitions(val scalaVersionOpt: Option[String])
       expect(output == message)
     }
   }
+  test("resource directory for coursier bootstrap launcher") {
+    val fileName = "hello.sc"
+    val message  = "1,2,3"
+    val inputs = TestInputs(
+      Seq(
+        os.rel / fileName ->
+          s"""|// using resourceDir "."
+              |import scala.io.Source
+              |
+              |val inputs = Source.fromResource("input").getLines.toSeq
+              |println(inputs.mkString)
+              |""".stripMargin,
+        os.rel / "input" -> message
+      )
+    )
+    inputs.fromRoot { root =>
+      os.proc(TestUtil.cli, "package", extraOptions, ".").call(
+        cwd = root,
+        stdin = os.Inherit,
+        stdout = os.Inherit
+      )
+
+      val outputName = if (Properties.isWin) "app.bat" else "app"
+      val launcher   = root / outputName
+
+      val output = os.proc(launcher.toString).call(cwd = root).out.text().trim
+      expect(output == message)
+    }
+  }
+
+  def libraryWithResourceTest(): Unit = {
+    val fileName     = "MyLibrary.scala"
+    val outputLib    = "my-library.jar"
+    val resourceFile = "input"
+    val inputs = TestInputs(
+      Seq(
+        os.rel / fileName ->
+          s"""|// using resourceDir "."
+              |
+              |class MyLibrary {
+              |  def message = "Hello"
+              |}
+              |""".stripMargin,
+        os.rel / resourceFile -> "1,2,3"
+      )
+    )
+    inputs.fromRoot { root =>
+      os.proc(TestUtil.cli, "package", extraOptions, ".", "-o", outputLib, "--library").call(
+        cwd = root,
+        stdin = os.Inherit,
+        stdout = os.Inherit
+      )
+
+      val output = os.proc("jar", "tf", root / outputLib).call(cwd = root).out.text().trim
```

Comment:
The output file could be inspected by `java.util.zip.ZipFile` instead. Something like
```scala
import java.util.zip.ZipFile
val zf = new ZipFile((root / outputLib).toIO)
val entry = zf.getEntry(resourceFile)
expect(entry != null)
```